dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -278,8 +278,14 @@ object Datadog {
         // so some things may yet not be initialized -> not accessible, some things may already be
         // initialized and be not mutable anymore
         additionalConfiguration[DD_SOURCE_TAG]?.let {
-            if (it.toString().isNotBlank()) {
-                CoreFeature.sourceName = it.toString()
+            if (it is String && it.isNotBlank()) {
+                CoreFeature.sourceName = it
+            }
+        }
+
+        additionalConfiguration[DD_SDK_VERSION_TAG]?.let {
+            if (it is String && it.isNotBlank()) {
+                CoreFeature.sdkVersion = it
```

Comment:
I am wondering as a follow up on @xgouchet question. Can this parameter be passed a an `Int` here somehow from react-native ?

Reply:
We are following semantic versioning, so it will be always as `x.y.z` in the string format imo. Don't see at what point it can be changed.